src/main/java/com/aws/greengrass/util/platforms/unix/LinuxPlatform.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.util.platforms.unix;
+
+import org.zeroturnaround.process.PidUtil;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static com.aws.greengrass.util.Utils.inputStreamToString;
+
+public class LinuxPlatform extends UnixPlatform {
+    private static final String MEMORY_KEY = "Memory";
+    private static final String CGROUP_MEMORY_LIMITS = "memory.limit_in_bytes";
+    private static final String CPU_KEY = "CPU";
+    private static final String UNICODE_SPACE = "\\040";
+    private static final List<LinuxCgroup> ENABLED_CGROUPS = Arrays.asList(LinuxCgroup.Memory, LinuxCgroup.CPU);
+
+    public void createResourceManager(String componentName) {
+        try {
+            Set<String> mounts = getMountedPaths();
+            if (!mounts.contains(LinuxCgroup.getRootPath())) {
+                runCmdWithRoot(LinuxCgroup.rootMountCmd());
+                for (LinuxCgroup cg : ENABLED_CGROUPS) {
+                    Files.createDirectory(cg.getSubsystemRootPath());
+                }
+            }
+
+            for (LinuxCgroup cg : ENABLED_CGROUPS) {
+                if (!mounts.contains(cg.getSubsystemRootPath())) {
+                    runCmdWithRoot(cg.subsystemMountCmd());
+                }
+                if (!Files.exists(cg.getSubsystemGGPath())) {
+                    Files.createDirectory(cg.getSubsystemGGPath());
+                }
+                if (!Files.exists(cg.getSubsystemComponentPath(componentName))) {
+                    Files.createDirectory(cg.getSubsystemComponentPath(componentName));
+                }
+            }
+        } catch (IOException e) {
+            logger.atError().setCause(e).kv("componentName", componentName)
+                    .log("Failed to create system resource manager");
+        } catch (InterruptedException e) {
```

Comment:
Is silently failing to create cgroup the right failure handling behavior? Should the component still be started up in this case?